presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/DomainConverter.java

```diff
@@ -52,14 +48,7 @@ private static Domain translateDomain(Domain domain)
         ValueSet valueSet = domain.getValues();
         Type type = domain.getType();
         if (type instanceof TimestampType || type instanceof TimestampWithTimeZoneType || type instanceof TimeType || type instanceof TimeWithTimeZoneType) {
-            if (valueSet instanceof EquatableValueSet) {
```

Comment:
Let's throw an exception if we get this (we shouldn't but be safe)

Reply:
Done in just now in a forced push.